earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -45,22 +45,25 @@ class StorageManager
     using HandleType = khRefGuard<AssetType>;
     using AssetKey = SharedString;
 
-    StorageManager(uint cacheSize, const std::string & type) :
+    StorageManager(uint cacheSize, bool enabled, uint64 maxMemory, const std::string & type) :
         cache(cacheSize),
-        assetType(type) {}
+        assetType(type) { SetCacheMemoryLimit(enabled, maxMemory); }
     ~StorageManager() = default;
 
     inline uint32 CacheSize() const { return cache.size(); }
-    inline uint32 CacheCapacity() const { return cache.capacity(); }
+    inline uint64 CacheCapacity() const { return cache.capacity(); }
     inline uint32 DirtySize() const { return dirtyMap.size(); }
+    inline uint64 CacheMemoryUse() const { return cache.getMemoryUse(); }
+    inline void SetCacheMemoryLimit(bool enabled, uint64 maxMemory) { cache.setCacheMemoryLimit(enabled, maxMemory); }
```

Comment:
Minor: I don't know that it makes sense for this function to be public, but perhaps it's necessary for testing.

Reply:
SetCacheMemoryLimit is used by the PurgeCacheBasedOnMemoryUtilization unit test to verify that objects are pruned from the cache if the total memory used by the cache exceeds the set limit.